File: Population.java
import java.util.Vector;

/**
 * Class representing a population of individuals
 */
public class Population extends Vector<Individual> {
	private Map map;
		
	/**
	 * Actual standard ctor.
	 * @param map The map.
	 * @param initialSize The initial size of the population.
	 */
	Population(Map map, int initialSize) {
		for(int i = 0; i < initialSize; ++i)
		{
			add(new Individual(map));
		}
	}
	
	/**
	 * Standard ctor.
	 * @param map The map.
	 */
	public Population(Map map) {
		this(map, 0);
	}
	
	/**
	 * Randomly selects an individual out of the population
	 * proportionally to its fitness.
	 * @return The selected individual.
	 */
	Individual randomSelection() {
		// TODO implement random selection
		// an individual should be selected with a probability
		// proportional to its fitness
		int sumfitness=0;
		for(int i=0;i<size();i++){
			sumfitness+=get(i).getFitness();
		}
		double r=Math.random()*(sumfitness-1);
		int sumprevious=0;
		for (int i=0;i<size();i++){
			if(r>=sumprevious&&r<(sumprevious+get(i).getFitness())){
				//System.out.println("class Population 46");
				return get(i);
			}
			sumprevious+=get(i).getFitness();
		}
		return null;
	}
	
}

File: Population.py
#Class representing a population of individuals
import random

from Individual import Individual

class Population:
    def __init__(self,map,initialSize):
        self.vector=[]
        for i in range(initialSize):
            self.vector.append(Individual(map))

    def randomSelection(self):
        # TODO implement random selection
        # an individual should be selected with a probability
        # proportional to its fitness
        sumfiness=0
        for i in range(len(self.vector)):
            sumfiness+=self.vector[i].fitness
        fitness_list=[]
        for i in range(len(self.vector)):
            fitness_list.append(float(self.vector[i].fitness)/float(sumfiness))
        r=random.choices(self.vector,weights=fitness_list,k=2)
        while r[0]==r[1] :
            r = random.choices(self.vector, weights=fitness_list, k=2)
        return r[0],r[1]